From 2198458190c73a9135ce6548963fdb65fca7c9f3 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 22 Mar 2020 23:31:43 +0200
Subject: [PATCH] [SW-0] ppboot: run udelay_init() before udelay()

This will keep the udelay() correctness in any case and on any CPU
frequency.

Change-Id: I4d5434208b9f3c185b5a8e533458e7977522b9ff
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/app/ppboot.c b/app/ppboot.c
index cc67e48..c1b7e1d 100644
--- a/app/ppboot.c
+++ b/app/ppboot.c
@@ -15,6 +15,7 @@
 #include "preboot.h"
 #include "uart.h"
 #include "spi.h"
+#include "pll.h"
 
 /* Stable ID can be 0 or 1, if not defined return primary offset (0) */
 static u32 next_image_offset(u32 offset, char *name)
@@ -135,6 +136,12 @@
 
 	hl_plat_init();
 	hl_jtag_init();
+	/*
+	 * Note: we assume CPU frequency is not changed while CPU is running,
+	 * so the udelay init is done once before we start.
+	 */
+	udelay_init();
+
 	preboot_sync_btl();
 	hl_uart_init();
 	ppboot_pcie_init();
